src/cow-react/modules/swap/containers/NewSwapWidget/propsChecker.ts

```diff
@@ -60,5 +60,24 @@ export function swapPagePropsChecker(prev: SwapFormProps, next: SwapFormProps):
 }
 
 export function genericPropsChecker(prev: any, next: any): boolean {
-  return JSON.stringify(prev) === JSON.stringify(next)
+  if (typeof prev === 'object') {
+    const prevKeys = Object.keys(prev)
+    const nextKeys = Object.keys(next)
+    // Just in case, we take the longest list of keys
+    const keys = prevKeys.length > nextKeys.length ? prevKeys : nextKeys
```

Comment:
its not easier to just return false earlier if they have diferent number of elements?

Reply:
In fact it's almost impossible, added the check just in case

Reply:
mmm, why, if the props have some optional propst would make sense, right? (if this is meant to be generic for any props passed)

Reply:
>just return false earlier if they have diferent number of elements

Done this :) It's much simplier